src/formatDate.js

```diff
@@ -9,6 +9,64 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const oldDateSeparator = fromFormat[3];
+  const splitFromFormat = date.split(oldDateSeparator);
+  const newFormatDate = [];
+  const year = [];
+  let month = '';
+  let day = '';
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    switch (fromFormat[i][0]) {
+      case 'Y':
+        year.push(splitFromFormat[i]);
+        break;
+
+      case 'M':
+        month = splitFromFormat[i];
+        break;
+
+      case 'D':
+        day = splitFromFormat[i];
+        break;
+
+      default:
+        break;
```

Comment:
The switch statement is using the first character of the format string to determine the type of the date part. This could lead to issues if the format string is not as expected (e.g., lowercase 'y' instead of 'Y'). It would be more robust to compare the entire string to 'YYYY', 'YY', 'MM', 'DD'.